File: copy-files.js
const fs = require('fs');
const path = require('path');

const src = 'f:\\my work\\programming\\vercel\\ai-tools-update';
const dst = 'F:\\temp-repo-clone';

const excludeDirs = new Set(['.git', 'node_modules', 'dist', '.astro', '.vscode', '__pycache__', '-p']);
const excludeFiles = new Set(['copy-files.js', 'copy_files.py', 'temp-copy.bat', 'copy-to-repo.ps1', 'copy.ps1']);

function copyRecursive(srcPath, dstPath) {
    const entries = fs.readdirSync(srcPath, { withFileTypes: true });
    let copied = 0;

    for (const entry of entries) {
        if (excludeDirs.has(entry.name) || excludeFiles.has(entry.name)) {
            continue;
        }

        const srcFile = path.join(srcPath, entry.name);
        const dstFile = path.join(dstPath, entry.name);

        if (entry.isDirectory()) {
            if (!fs.existsSync(dstFile)) {
                fs.mkdirSync(dstFile, { recursive: true });
            }
            copied += copyRecursive(srcFile, dstFile);
        } else {
            fs.copyFileSync(srcFile, dstFile);
            copied++;
            if (copied % 100 === 0) {
                console.log(`Copied ${copied} files...`);
            }
        }
    }

    return copied;
}

console.log(`Copying from ${src} to ${dst}...`);
const totalCopied = copyRecursive(src, dst);
console.log(`\nCopy complete! ${totalCopied} files copied.`);

// List sample files
const files = fs.readdirSync(dst).slice(0, 10);
console.log('\nSample files in destination:');
files.forEach(f => console.log(`  - ${f}`));

File: copy_files.py
import shutil
import os
from pathlib import Path

src = r"f:\my work\programming\vercel\ai-tools-update"
dst = r"F:\temp-repo-clone"

exclude_dirs = {'.git', 'node_modules', 'dist', '.astro', '.vscode', '__pycache__'}
exclude_files = {'copy_files.py', 'temp-copy.bat'}

def should_exclude(path, src_root):
    rel_path = os.path.relpath(path, src_root)
    path_parts = Path(rel_path).parts
    return any(ex in path_parts for ex in exclude_dirs)

print(f"Copying from {src} to {dst}...")
copied = 0

for root, dirs, files in os.walk(src):
    # Remove excluded directories from dirs list to prevent walking into them
    dirs[:] = [d for d in dirs if d not in exclude_dirs]
    
    if should_exclude(root, src):
        continue
    
    # Calculate relative path and create destination directory
    rel_path = os.path.relpath(root, src)
    if rel_path == '.':
        dst_dir = dst
    else:
        dst_dir = os.path.join(dst, rel_path)
    
    os.makedirs(dst_dir, exist_ok=True)
    
    # Copy files
    for file in files:
        if file in exclude_files:
            continue
        
        src_file = os.path.join(root, file)
        dst_file = os.path.join(dst_dir, file)
        
        try:
            shutil.copy2(src_file, dst_file)
            copied += 1
            if copied % 100 == 0:
                print(f"Copied {copied} files...")
        except Exception as e:
            print(f"Error copying {src_file}: {e}")

print(f"\nCopy complete! {copied} files copied.")

# List some files in destination
print("\nSample files in destination:")
dst_files = list(Path(dst).glob('*'))[:10]
for f in dst_files:
    print(f"  - {f.name}")
